### ISSUE TITLE
cppcheck reports warnings at wrong locations

### ISSUE THREAD

**Issue by Hiftu** [johnthagen/clion-cppcheck#4]
### Environment

* Operating System (e.g. Ubuntu 16.04 x64): RHEL 7.1 x86_64
* IDE Version (e.g. CLion 2016.3.2): clion 2016.3
* Cppcheck executable version (`cppcheck --version`): 
* Cppcheck plugin version:  Cppcheck 1.75
* Exact strings used in cppcheck plugin options: 
  * **cppcheck path**: /usr/bin/cppcheck
  * **cppcheck options**:  --enable=warning,performance,portability,style --language=c++ --std=c++11

### Expected behaviour
Warnings from cppcheck plugin appears only for the right file.

### Actual behaviour
If a header contains an issue (lets say at line 18) then this issue appear in every files (at line 18) which includes this header.

### Steps to reproduce the behaviour
Create a header file with a variable without initialization.
```
Test.h
int x;
```

Include this header.
```
Test.cc

#include "Test.h"

int main()
{
  return x;
}

```

### Possible solution:
Change regexp to capture filename. Get name of the current document.
```
   Pattern pattern = Pattern.compile("^\\[(.+):(\\d+)\\]:\\s+\\((\\w+)\\)\\s+(.+)");
   final String documentName = file.getVirtualFile().getName();
```
Use these information to filter out unmatching files.
```
   final String filename = matcher.group(1);
   if( !filename.endsWith(documentName) ) {
       continue;
   }
   int lineNumber = Integer.parseInt(matcher.group(2), 10);
   final String severity = matcher.group(3);
   final String errorMessage = matcher.group(4);
```

**Comment by Hiftu** [comment]
Correct versions:
* executable version: Cppcheck 1.75
* plugin version: 1.0.6

**Comment by johnthagen** [comment]
@Hiftu Yep you're right, hadn't seen that before. Thanks for the report and suggested fix. I'll see what I can do.

**Comment by johnthagen** [comment]
@Hiftu Please update to 1.0.7 and see if that solves your problem.

**Comment by johnthagen** [comment]
Fixed in c948b52db7102e839be6c70cc94311c4dcbe39e7.
